src/metadata/debug_utils.rs

```diff
@@ -252,6 +293,16 @@ impl DebugUtils {
                 );
             }
         }
+
+        if self.active_map.contains(&DebugBinding::PrintFelt252) {
+            dbg!("registering felt252 print");
+            unsafe {
+                engine.register_symbol(
+                    "__debug__print_felt252",
+                    print_pointer_felt252 as *const fn([u32; 4]) -> () as *mut (),
```

Comment:
The first cast should have the same signature as the function.
```suggestion
                    print_pointer_felt252 as *const fn(u64, u64, u64, u64) -> () as *mut (),
```